jobpy/__main__.py

```diff
@@ -63,19 +63,34 @@ def main():
                  'SEARCHTERMS_PATH' : default_args['SEARCHTERMS_PATH'],
                  })
 
-    # init class
-    jobpy_obj = jobpy(args)
-
-    # parse the xslx to filter list, scrape new listings & add to master xslx
-    jobpy_obj.masterlist_to_filterjson()
-    if not args['NO_SCRAPE'] :
-        jobpy_obj.daily_scrape_dict = {}
-        scrape_indeed_to_pickle(jobpy_obj)
-        scrape_monster_to_pickle(jobpy_obj)
-        scrape_glassdoor_to_pickle(jobpy_obj)
-    jobpy_obj.pickle_to_masterlist()
-
-    print ("done.\nsee un-archived jobs in " + args['MASTERLIST_PATH'])
+    # init class + logging
+    jp = JobPy(args)
+    jp.init_logging()
+
+    # parse the masterlist_path to update filter list
+    jp.masterlist_to_filterjson()
+
+    # get jobs by either scraping jobs or loading today's dumped pickle
+    if args['NO_SCRAPE']:
+        jp.load_pickle()
+    else:
+        # @TODO pass more data via JobPy init args
+        for provider in (Indeed(args), Monster(args), GlassDoor(args)):
+            try:
+                provider.scrape()
+                jp.scrape_data.update(provider.scrape_data)
+            except Exception as e:
+                jp.logger.error('failed to scrape {}: {}'.format(
+                    provider.__class__.__name__, str(e)))
```

Comment:
consider `f-strings` instead of `''.format()`:

`jp.logger.error(f'failed to scrape {provider.__class__.__name__}: {str(e)}')`

Reply:
I didn't even know f-strings existed.  Looks way better.  It hurts my eyes finding which {number} corresponds to which .format(zero, one, two)